[Michael Lee] iP

michael-lee-sk opened the pull request:
**Duke**
> Be regular and orderly in your life, that you may be violent and original in your work. - Clive Barker [Source](https://www.brainyquote.com/quotes/gustave_flaubert_109857)

Duke creates your lists for you, so you do not have to
* text-based
* *SUPER* easy to learn
* ~FAST~ SUPER FAST

All you need to do is,
1. Download it here for a low price of $0
2. let it manage your tasks for you 😉

Features include
- [x] Managing tasks

Commit:
Add Gradle support

michael-lee-sk committed:
Level 1. Greet, Echo, Exit

michael-lee-sk committed:
Level 2. Add, List

michael-lee-sk committed:
Level 2. Add, List

michael-lee-sk committed:
Level 3. Mark as Done

michael-lee-sk committed:
Level 4. ToDos, Events, Deadlines

michael-lee-sk committed:
A-TextUiTesting

michael-lee-sk committed:
Level 5. Handle Errors

michael-lee-sk committed:
Level 5. Handle Errors

michael-lee-sk committed:
updated A-TextUiTesting

michael-lee-sk committed:
A-Enums

michael-lee-sk committed:
A-Enums (updated)

michael-lee-sk committed:
Implemented level 7 - save

michael-lee-sk committed:
level 8 implementation - date and time

michael-lee-sk committed:
level 7 updated

michael-lee-sk committed:
level 7 updated again

michael-lee-sk committed:
level 8 updated

michael-lee-sk committed:
Level 7 - save and merge

michael-lee-sk committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java

michael-lee-sk committed:
implemented and merged level 8

michael-lee-sk committed:
A-MoreOOP

michael-lee-sk committed:
A-Packages

michael-lee-sk committed:
A-JUnit

michael-lee-sk committed:
A-JavaDoc

michael-lee-sk committed:
Level-9

michael-lee-sk committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/duke/DeleteException.java
#	src/main/java/duke/Duke.java
#	src/main/java/duke/DukeException.java
#	src/main/java/duke/Parser.java
#	src/main/java/duke/Ui.java

michael-lee-sk committed:
Merged branch "Level-9"

michael-lee-sk committed:
A-CodingStandard

michael-lee-sk committed:
Merge branch 'branch-A-CodingStandard'

michael-lee-sk committed:
Merged branch 'branch-A-CodingStandard'

michael-lee-sk committed:
A-JavaDoc

michael-lee-sk committed:
Merge branch 'branch-A-JavaDoc'

michael-lee-sk committed:
Merged branch 'branch-A-JavaDoc'

Ongzl commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,172 @@
+     * @throws DeleteException If delete is incomplete.
+     * @throws IOException If an input or output operation is failed or interpreted.
+     */
+    public void parseCommand() throws DukeException, DeleteException, IOException, FindException {
+
+        Activity activity;
+        if (command.equals("bye")) {
```
This method / while loop might be too long. This [coding quality website](https://nus-cs2103-ay2122s1.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods)  stated the following:

>Be wary when a method is longer than the computer screen, and take corrective action when it goes
>beyond 30 LOC (lines of code). The bigger the haystack, the harder it is to find a needle.

This is a little longer than 30 LOC , though I too am not sure how to condense this part of logic haha, maybe implementing a 'command' class?

Xiaoyunnn replied:
I agree with @Ongzl. Perhaps, you can consider moving the content in the switch cases to separate methods to shorten this method body?

michael-lee-sk replied:
hmm, may i clarify something? 30 LOC is for one method right? Technically I have many different switch cases, by which each case has << 30 LOC (the longest switch case has 15 LOC inclusive of empty lines). So a method refers to one switch case right, not the entire switch?

Ongzl replied:
sftlr, I am not too sure too because the site itself is quite vague, but I think as long as measures are put in to minimize the size of the method then it should be fine.

Ongzl commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,126 @@
+public class Storage {
+    private static String filePath;
+
+    private boolean exit = false;
```
I think the boolean variable should be name to something that sounds like booleans? Maybe shouldExit or something?

michael-lee-sk replied:
Thanks for pointing this out, will do the changes!

Ongzl commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,79 @@
+     * @param dateAndTime Date and time of the event.
+     */
+    public Event(String description, String dateAndTime) {
+        super(description);
+        this.description = description;
```
I am not sure if this double description assignment is necessary since there is a description variable for Task already and the super(description) already assigns the description

Xiaoyunnn replied:
I agree with @Ongzl , you can still get the subclass instance's `description` (from parent class) without the subclass having a private field for `description`.

michael-lee-sk replied:
Thanks for pointing this out, will do the changes!

Ongzl commented on src/main/java/duke/Event.java:
```diff
@@ -0,0 +1,79 @@
+ * "Event <event name> /at <event date>".
+ */
+public class Event extends Task{
+    protected String description;
```
I am not sure if having two String description for the children of Task is a good idea since it looks like you are assigning description twice and have 2 description variables. I think you did the same for Event, Deadline and ToDo, I am not sure if this is wise.

Ongzl reviewed (commented):
Coding standard looks good to me. I am not sure if the double description thing is under coding standards or not, but just thought to let you know, cheers

Xiaoyunnn commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,79 @@
+    protected String description;
+    protected boolean isDone;
+
+    /**
+     * Public constructor for a task.
+     *
+     * @param description Description of the event.
+     */
+    public Task(String description) {
+        this.description = description;
+        this.isDone = false;
+    }
```
I think this JavaDoc should begin with a verb? e.g. "Instantiates" or "Constructs". I believe this applies for your other constructors as well.

michael-lee-sk replied:
Thanks for pointing this out, did not know this haha. Will do the changes accordingly :)

Xiaoyunnn commented on src/main/java/duke/Task.java:
```diff
@@ -0,0 +1,79 @@
+    }
+
+    /**
+     * Set boolean if task is done, or not.
```
I think this line should start with "Sets"?

Xiaoyunnn commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,172 @@
+        return !storage.isExit();
+    }
+
+    /**
+     * Parses the user command and inputs the task accordingly.
+     *
+     * @throws DukeException If task input by user is incomplete.
+     * @throws DeleteException If delete is incomplete.
+     * @throws IOException If an input or output operation is failed or interpreted.
+     */
+    public void parseCommand() throws DukeException, DeleteException, IOException, FindException {
```
I think you missed out the description for `FindException`?

michael-lee-sk committed:
Updated according to PR comments

michael-lee-sk committed:
Merge branch 'add-gradle-support' of https://github.com/michael-lee-sk/ip into branch-A-Gradle

michael-lee-sk committed:
branch-A-Gradle

michael-lee-sk committed:
implemented A-Checkstyle

michael-lee-sk committed:
A-Checkstyle with xml files

michael-lee-sk committed:
Level-10 commit 1

michael-lee-sk committed:
Level-10 commit 2

michael-lee-sk committed:
Level-10 commit 3

michael-lee-sk committed:
Merge branch 'branch-Level-10'

michael-lee-sk committed:
Merged 'branch-Level-10'

michael-lee-sk committed:
updated build.gradle for Jar file

michael-lee-sk committed:
Asserts will enable us to test for the correctness of the code based on assumptions that will never be wrong, helping us to spot for errors.

Lets insert certain Assert statements in various places, to ensure the assumptions hold up throughout the code.

michael-lee-sk committed:
*Refactor code to match code quality standards*
Code does contain certain violations of the code quality standards.

This could reduce the readability  of the code, and the general code quality.

Lets refactor code that may violate the code quality standards.

michael-lee-sk committed:
Merge pull request #1 from michael-lee-sk/branch-A-Assertions

Add Asserts

michael-lee-sk committed:
*Refactor code to match code quality standards*
Code does contain certain violations of the code quality standards.

This could reduce the readability  of the code, and the general code quality.

Lets refactor code that may violate the code quality standards.

michael-lee-sk committed:
Clean Javadocs

michael-lee-sk committed:
Merge pull request #2 from michael-lee-sk/branch-A-CodeQuality

A-CodeQuality

michael-lee-sk committed:
Add Delete and Done classes, with GeneralCommand interface

michael-lee-sk committed:
Update done on Delete and Done classes.
Add DoneException

michael-lee-sk committed:
Undo commit 1

michael-lee-sk committed:
Undo commit 2
Added done

michael-lee-sk committed:
Undo commit 3
Completed Done undo, but slightly buggy > 2 dones

michael-lee-sk committed:
Undo commit 4
Slight changes to done, still buggy

michael-lee-sk committed:
Merged C-Undo and updated

michael-lee-sk committed:
Update README.md

michael-lee-sk committed:
A-BetterGui

michael-lee-sk committed:
Merge remote-tracking branch 'origin/master'

michael-lee-sk committed:
A-BetterGui 2

michael-lee-sk committed:
updated todo

michael-lee-sk committed:
updated event

michael-lee-sk committed:
updated deadline

michael-lee-sk committed:
updated find

michael-lee-sk committed:
general updates on parsing of dates

michael-lee-sk committed:
Set theme jekyll-theme-merlot

michael-lee-sk committed:
ui.png

michael-lee-sk committed:
new Gui 1

michael-lee-sk committed:
updated new Gui, added new Ui.PNG

michael-lee-sk committed:
updated new Gui, finalised Ui.PNG and debugged

michael-lee-sk committed:
cleaned up code

michael-lee-sk committed:
renamed Ui.PNG to Ui.png

michael-lee-sk committed:
Final code

michael-lee-sk committed:
A-UserGuide

michael-lee-sk committed:
Update README.md

michael-lee-sk committed:
Update README.md

michael-lee-sk committed:
Final final code

michael-lee-sk committed:
updated commas

michael-lee-sk committed:
Update README.md
